### PR TITLE
Batch normalization dtype depends on inputs dtype

### PR DESCRIPTION
**Pull request by Gara-Dorta**
Make the batch normalization layer dtype be the same type as the inputs,
before it was always float32

### CODE REVIEW

**Comment by caisq** [comment]
ping @fchollet

**Comment by fchollet** [comment]
Thank you for your patience.

Weight `dtype` is already inferred from inputs dtype (unless explicitly specified in the layer constructor) in the base layer's `__call__`, and thus this PR is not warranted. See: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/layers/base.py#L598

As a general feedback point for future PRs: when submitting PRs for such fixes, please include unit tests (a test that would fail at `master` but would pass with the proposed fix).
